src/awkward/_v2/_reducers.py

```diff
@@ -44,14 +44,22 @@ class ArgMin(Reducer):
 
     @classmethod
     def apply(cls, array, parents, outlength):
+        dtype = (
+            np.dtype(np.int64)
+            if (
+                str(array.form.type).startswith("datetime64")
+                or str(array.form.type).startswith("timedelta64")
+            )
```

Comment:
We should not be using the string representations of objects to determine what type they are. Currently, it would be

```python
isinstance(t, ak._v2.types.NumpyType) and t.primitive.startswith("datetime64")
```

but this should be in a helper function. Hold on a moment, I'll fix this FIXME:

https://github.com/scikit-hep/awkward-1.0/blob/f795658559d61f1fd394c828e95608c9de8ee027/src/awkward/_v2/types/numpytype.py#L14-L65

(In the original design, I hadn't counted on datetimes and timedeltas having units, much less _arbitrary_ units, so I've got to turn that dict into a function. I'll do it now as a tiny PR.)

Reply:
PR #1146 will merge as soon as its tests pass, but it provides functional access to this information. The check (which you can put into a helper function) would then be

```python
(
    isinstance(t, ak._v2.types.NumpyType)
    and ak._v2.types.numpytype.primitive_to_dtype(t.primitive).kind.upper() == "M"
)
```

Reply:
Done in 883f0bb3e9ef277e0d6513521b700cac0ae963eb (not yet checked in).